core/common/src/main/java/alluxio/metrics/MetricsSystem.java

```diff
@@ -158,6 +164,37 @@ public static void startSinks(String metricsConfFile) {
     startSinksFromConfig(config);
   }
 
+  /**
+   * Initialize the {@link MetricsSystem}.
+   * This should be called after setting the {@link CommonUtils#PROCESS_TYPE}
+   * and before creating any metric in the current process.
+   */
+  public static void init() {
```

Comment:
Could you make this synchronized and have it be a no-op if `sSourceName` has already been set? The reason is the client may make multiple calls to init when multiple filesystem objects are created (ie. for different users).
Also, where is this called for a client?

Reply:
Sounds good. I maked it a no-op if already been set and maked it synchronized.
Added the call of init() to FileSystemContext.initContext() and a bunch of unit tests.